Common/GltfExport.cs

```diff
@@ -0,0 +1,182 @@
+﻿using GaneshaDx.Resources.ContentDataTypes.Polygons;
+using System.Numerics;
+using System.Text.Json;
+using SharpGLTF.Geometry;
+using SharpGLTF.Materials;
+using SharpGLTF.Memory;
+using SharpGLTF.Schema2;
+using SharpGLTF.Geometry.VertexTypes;
+using System.IO;
+using GaneshaDx.Resources;
+using System.Collections.Generic;
+using GaneshaDx.Environment;
+using Microsoft.Xna.Framework;
+using Vector4 = System.Numerics.Vector4;
+using Vector3 = System.Numerics.Vector3;
+using Vector2 = System.Numerics.Vector2;
+using MTex2D = Microsoft.Xna.Framework.Graphics.Texture2D;
+
+namespace GaneshaDx.Common
+{
+  public static class GltfExport {
+
+		/**
+		* Exports the map model to the .gltf format.
+		*/
+		public static void Export(string filePath) {
+			var stateTexture = CurrentMapState.StateData.Texture;
+			
+			var greyPalette = new List<Color>() {
+				Utilities.GetColorFromHex("000000"),
+				Utilities.GetColorFromHex("111111"),
+				Utilities.GetColorFromHex("222222"),
+				Utilities.GetColorFromHex("333333"),
+				Utilities.GetColorFromHex("444444"),
+				Utilities.GetColorFromHex("555555"),
+				Utilities.GetColorFromHex("666666"),
+				Utilities.GetColorFromHex("777777"),
+				Utilities.GetColorFromHex("888888"),
+				Utilities.GetColorFromHex("999999"),
+				Utilities.GetColorFromHex("AAAAAA"),
+				Utilities.GetColorFromHex("BBBBBB"),
+				Utilities.GetColorFromHex("CCCCCC"),
+				Utilities.GetColorFromHex("DDDDDD"),
+				Utilities.GetColorFromHex("EEEEEE"),
+				Utilities.GetColorFromHex("FFFFFF")
+			};
+			
+			var texturedMesh = new MeshBuilder<VertexPosition, VertexTexture1>("texturedMesh");
+			var texturedPrims = new List<PrimitiveBuilder<MaterialBuilder, VertexPosition, VertexTexture1, VertexEmpty>>();
```

Comment:
I try to avoid abbreviations for the most part. Lets rename this to `texturedPrimatives`